GeneratorInterface/HydjetInterface/src/HydjetHadronizer.cc

```diff
@@ -107,8 +102,14 @@ HydjetHadronizer::HydjetHadronizer(const ParameterSet& pset)
   maxEventsToPrint_ = pset.getUntrackedParameter<int>("maxEventsToPrint", 0);
   LogDebug("Events2Print") << "Number of events to be printed = " << maxEventsToPrint_;
 
-  if (embedding_)
-    src_ = pset.getParameter<edm::InputTag>("backgroundLabel");
+  if (embedding_) {
+    cflag_ = 0;
+    src_ = mayConsume<HepMCProduct>(
+        pset.getUntrackedParameter<edm::InputTag>("backgroundLabel", edm::InputTag("generator", "unsmeared")));
```

Comment:
Shouldn't this parameter stay as tracked? It affects physics and therefore should to be included in the event provenance.

Reply:
@makortel It does not included to the gen fragment by default, so to avoid changing all configs I added default value. Anyway for current underlying sample it can't be various. In case of different label (then underlying sample has) the generation fails.